modules/custom_operations/user_ie_extensions/fft.cpp

```diff
@@ -0,0 +1,369 @@
+            cvScale(out, out, 1.0 / sqrtf(channels * rows), 0);
+
+            if (centered)
+                fftshift(out, false);
+
+            CvMat out_col_header, *out_col;
+            out_col = cvReshape(out, &out_col_header, 2, channels * rows);
+
+            cvCopy(out_col, outSlice, 0);
+
+            cvReleaseData(inp_col);
+            cvReleaseMat(&inp_col);
+
+            cvReleaseData(out);
+            cvReleaseMat(&out);
+
+            cvReleaseMat(&inpSlice);
+            cvReleaseMat(&outSlice);
+        });
+    } else if (dims.size() == 5 && numSignalDims == 2 && signalDimsData[0] == 2 && signalDimsData[1] == 3) {
+        const int channels = dims[1];
+        int rows = dims[2];
+        int cols = dims[3];
+        int planeSize = rows * cols * 2;  // 2 is last dimension size
+        InferenceEngine::parallel_for(batch * channels, [&](size_t d) {
+            CvMat* inp = cvCreateMatHeader(rows, cols, CV_32FC2);
+            CvMat* out = cvCreateMatHeader(rows, cols, CV_32FC2);
+            cvSetData(inp, reinterpret_cast<void*>(inpData + d * planeSize), cols * 2 * sizeof(float));
+            cvSetData(out, reinterpret_cast<void*>(outData + d * planeSize), cols * 2 * sizeof(float));
+
+            if (centered)
+                fftshift(inp, true);
+
+            if (inverse)
+                cvDFT(inp, out, CV_DXT_INVERSE, 0);
+            else
+                cvDFT(inp, out, CV_DXT_FORWARD, 0);
+            cvScale(out, out, 1.0 / sqrtf(cols * rows), 0);
+
+            if (centered)
+                fftshift(out, false);
+
+            cvReleaseMat(&inp);
+            cvReleaseMat(&out);
+        });
+    } else if (dims.size() == 4 && numSignalDims == 2 && signalDimsData[0] == 1 && signalDimsData[1] == 2) {
+        int rows = dims[1];
+        int cols = dims[2];
+        int planeSize = rows * cols * 2;  // 2 is last dimension size
+        InferenceEngine::parallel_for(batch, [&](size_t d) {
+            CvMat* inp = cvCreateMatHeader(rows, cols, CV_32FC2);
+            CvMat* out = cvCreateMatHeader(rows, cols, CV_32FC2);
+            cvSetData(inp, reinterpret_cast<void*>(inpData + d * planeSize), cols * 2 * sizeof(float));
+            cvSetData(out, reinterpret_cast<void*>(outData + d * planeSize), cols * 2 * sizeof(float));
+
+            if (centered)
+                fftshift(inp, true);
+
+            if (inverse)
+                cvDFT(inp, out, CV_DXT_INVERSE, 0);
+            else
+                cvDFT(inp, out, CV_DXT_FORWARD, 0);
+            cvScale(out, out, 1.0 / sqrtf(cols * rows), 0);
+
+            if (centered)
+                fftshift(out, false);
+
+            cvReleaseMat(&inp);
+            cvReleaseMat(&out);
+        });
+    } else if (dims.size() == 4 && numSignalDims == 1 && signalDimsData[0] == 1) {
+        int rows = dims[1];
+        int cols = dims[2];
+
+        const int planeSize = rows;
+        InferenceEngine::parallel_for(batch * cols, [&](size_t d) {
```

Comment:
CC @ilyachur 
Looks like we need to port this to API 2.0 to allow creation of optimized CPU kernels..

Reply:
@ilya-lavrenov could you please clarify what you mean

Reply:
I mean currently you have to use API 1.0 because we don't have such helpers in 2.0
we (OV team) need to think whether we need to move such helpers to 2.0 folder.

Reply:
We discussed this topic multiple times and at the current moment we don't have a plan to move this functionality to new extensions.
In order to support custom op user can add evaluate method.
For performance execution he can contribute to CPU plugin

Reply:
PR https://github.com/openvinotoolkit/openvino/pull/14485/ adds parallel_for for OV 2.0 API

Reply:
Thank you, I've updated